src/software/amazon/ion/impl/lite/IonSequenceLite.java

```diff
@@ -398,4 +400,313 @@ void writeBodyTo(IonWriter writer, SymbolTableProvider symbolTableProvider)
             writer.stepOut();
         }
     }
+
```

Comment:
this would be a nice spot for a comment about what `fromIndex` represents, particularly in the context of nested sublists.  might be nice/wise to mention how we know whether to throw a CME

Reply:
added javadoc to both inner class and `fromIndex`